tools/cmake/toolchains/GCC_ARM.cmake

```
@@ -66,24 +66,26 @@ function(mbed_set_toolchain_options target)
     )
 endfunction()
 
-# GCC ARM requires preprecessing linker script, execute generators to get definitions needed for
-# this step - linker options and compile definitions 
-function(mbed_generate_gcc_options_for_linker target definitions_file linker_options_file)
+# Generate a file containing compile definitions
+function(mbed_generate_gcc_options_for_linker target definitions_file)
     set(_compile_definitions 
         "$<TARGET_PROPERTY:${target},COMPILE_DEFINITIONS>"
     )
 
-    set(_linker_options
-        "$<TARGET_PROPERTY:${target},LINK_OPTIONS>"
+    # Remove macro definitions that contain spaces as the lack of escape sequences and quotation marks
+    # in the macro when retrieved using generator expressions causes linker errors.
+    # This includes string macros, array macros, and macros with operations.
+    # TODO CMake: Add escape sequences and quotation marks where necessary instead of removing these macros.
+    set(_compile_definitions
+       "$<FILTER:${_compile_definitions},EXCLUDE,.*[ ={ | =" | =\( ].*[ } | " | )]>"
```

To make it prettier:
```suggestion
       "$<FILTER:${_compile_definitions},EXCLUDE, +>"
```
As far as I experimented, it's enough to filter out spaces only. It's okay to have brackets, `|`, etc. in `compile_time_defs.txt` - it's referenced with `@` and no bash is involved.

---

Only `+` fails to build. Only + is not valid regex, is it?

---

Cmake error, working now ::)